File: frontend/src/types/candidate.ts
export interface Candidate {
  id: number;
  name: string;
  phone?: string | null;
  email?: string | null;
  position_name?: string | null;
  source?: string | null;
  education?: string | null;
  school?: string | null;
  major?: string | null;
  graduation_year?: string | null;
  work_years?: string | null;
  skills?: string[] | null;
  experience_summary?: string | null;
  resume_summary?: string | null;
  resume_parse_status?: string | null;
  latest_resume_id?: number | null;
  match_score?: number | null;
  match_level?: string | null;
  stage: string;
  owner?: string | null;
  last_followup_at?: string | null;
  created_at: string;
  updated_at: string;
}

File: backend/app/schemas/candidate.py
from datetime import datetime
from typing import Any, Dict, List, Optional

from pydantic import BaseModel, ConfigDict


class CandidateOut(BaseModel):
    id: int
    name: str
    phone: Optional[str] = None
    email: Optional[str] = None
    position_name: Optional[str] = None
    source: Optional[str] = None
    education: Optional[str] = None
    school: Optional[str] = None
    major: Optional[str] = None
    graduation_year: Optional[str] = None
    work_years: Optional[str] = None
    skills: Optional[List[str]] = None
    experience_summary: Optional[str] = None
    resume_summary: Optional[str] = None
    resume_parse_status: Optional[str] = None
    latest_resume_id: Optional[int] = None
    match_score: Optional[float] = None
    match_level: Optional[str] = None
    stage: str
    owner: Optional[str] = None
    last_followup_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
